### PR TITLE
Adds e2e tests

### PR DESCRIPTION
**Pull request by jaredjensen**
- Added e2e testing with Puppeteer, Jest, and MirageJS
- Added helpers to more easily inspect DOM and make assertions
- Fixed all linter warnings and tweaked config to avoid conflicts between ESLint and Prettier
- Added git hooks for tests
- Documented test strategy in README

### CODE REVIEW

File: e2e/site/init.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  // if (result.type === 'LoginSuccess') {
+  //   target.innerHTML = `Success:<br/>${result.getSessionToken()}<br/><br/>Logging out in 3 seconds...`;
+  //   setTimeout(async () => {
+  //     await forgerock.SessionManager.logout();
+  //     window.location.reload(true);
+  //   }, 3000);
+  // } else {
+  //   target.innerHTML = `Failure: ${result.getMessage()}`;
+  // }
+}
```
Comment (cerebrl) [inline e2e/site/init.js]: Should all of this be removed before merging?

Comment (jaredjensen) [reply e2e/site/init.js]: Yep, that one slipped through.

File: e2e/tests/confirmation-renderer.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+describe(testName, () => {
+  it('renders correctly', async () => {
+    if (expectedLabelText) {
+      const actualLabelText = await getInnerHtml(actualLabel);
+      expect(actualLabelText).toBe(expectedLabelText);
+    }
+
+    expect(actualButtons.length).toBe(expectedOptions.length);
+    for (let i = 0; i < expectedOptions.length; i++) {
```
Comment (cerebrl) [inline e2e/tests/confirmation-renderer.test.ts]: Just a bit nit-picky, but you could switch this to a `forEach` to simplify the logic just a tad.

File: e2e/tests/kba-create-renderer.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      // Ensure "custom" is visible when custom question is selected
+      const customQuestion = await getInnerHtml(actualOptions[actualOptions.length - 1]);
+      await setDropdownValue(actualDropdowns[i], customQuestion);
+      isHidden = await hasClass(actualCustomWraps[i], 'd-none');
+      expect(isHidden).toBe(false);
+      // Ensure "custom" is hidden when custom question is not selected
+      const firstQuestion = await getInnerHtml(actualOptions[0]);
+      await setDropdownValue(actualDropdowns[i], firstQuestion);
+      isHidden = await hasClass(actualCustomWraps[i], 'd-none');
+      expect(isHidden).toBe(true);
```
Comment (cerebrl) [inline e2e/tests/kba-create-renderer.test.ts]: This whole block is really dense and hard to read. A bit of space would do this wonders.

File: e2e/tests/confirmation-renderer.test.ts (hunk 2 above)
Comment (jaredjensen) [reply e2e/tests/confirmation-renderer.test.ts]: I used `forEach` originally, but switched to a normal `for` loop to avoid problems with `async`.

**Pull request merged**